### PR TITLE
Add Doppler integration (ECOINT-6)

### PR DESCRIPTION
**Pull request by nmanoogian**
### What does this PR do?

This PR creates the Doppler log collection integration. This is an API-based integration that uses OAuth to connect Doppler with Datadog. Once connected, Doppler will begin sending activity log data to Datadog.

### Motivation

What inspired you to submit this pull request?

### Review checklist

- [x] PR has a [meaningful title](https://github.com/DataDog/integrations-core/blob/master/CONTRIBUTING.md#pull-request-title) or PR has the `no-changelog` label attached
- [x] Feature or bugfix has tests (Not Relevant)
- [x] Git history is clean
- [x] If PR impacts documentation, docs team has been notified or an issue has been opened on the [documentation repo](https://github.com/DataDog/documentation/issues/new)
- [x] If this PR includes a log pipeline, please add a description describing the remappers and processors.

### CODE REVIEW

**Comment by neko-dd** [comment]
Created DOCS-8635 for docs review

File: doppler/README.md
Comment (michaelcretzman) [inline doppler/README.md:19]:
```suggestion
The integration will automatically send all Doppler Activity Logs to Datadog. No further configuration is available at this time.
```

File: doppler/README.md
Comment (michaelcretzman) [inline doppler/README.md:23]:
```suggestion
During installation, Doppler creates a test log to verify that the installed credentials work properly. Confirm that this test log is present in your logs to verify the installation.
```

File: doppler/README.md
Comment (michaelcretzman) [inline doppler/README.md:27]:
```suggestion
- Go to your [Doppler workplace settings][4] and disconnect the Datadog integration.
```

**Review by michaelcretzman — APPROVED** [review]
Approved with some minor edits.

File: doppler/README.md
Comment (emarsha94) [inline doppler/README.md:10]: Can you please add clear directions a user would take to install the integration? For example, go to the Doppler tile on the integrations page, click connect accounts, you will be redirected to doppler to log in, .. 

This should walk through the full oauth flow a user will take. I recommend testing the oauth flow and creating the steps on the steps you take!

File: doppler/assets/logs/doppler_tests.yaml
Comment (emarsha94) [inline doppler/assets/logs/doppler_tests.yaml:164]: please see here and follow steps above

File: doppler/assets/oauth_clients.json
Comment (emarsha94) [inline doppler/assets/oauth_clients.json:11]: please remove the agent_upgrade write scope from your oauth client and re-publish the client

File: doppler/manifest.json
Comment (emarsha94) [inline doppler/manifest.json:58]: this is missing the submitted data type. Please add this https://docs.datadoghq.com/developers/integrations/check_references/#classifier-tags

**Review by emarsha94 — CHANGES_REQUESTED** [review]

File: doppler/README.md
Comment (nmanoogian) [reply doppler/README.md:5]: Thanks! I've updated this section with more of those specifics.

File: doppler/README.md
Comment (nmanoogian) [reply doppler/README.md:13]: Absolutely, I re-tested the full oauth flow and provided more detail here.

File: doppler/assets/oauth_clients.json
Comment (nmanoogian) [reply doppler/assets/oauth_clients.json]: Done!

File: doppler/manifest.json
Comment (nmanoogian) [reply doppler/manifest.json]: I've added the logs datatype

**Comment by nmanoogian** [comment]
Thanks all for the feedback!

A question for you: Do I need the files added in the "WIP: Add tests" commit? I re-ran `ddev create` and noticed that these came up.

I still need to let CI run and collect the test log results. I'll push again once that's done.

**Comment by nmanoogian** [comment]
Another question: What should I provide for these manifest values if our integration doesn't produce metrics or provide service checks:

```
{
  "assets": {
    "integration": {
      "metrics": {
        "prefix": [
          "Missing data for required field."
        ],
        "metadata_path": [
          "Missing data for required field."
        ]
      },
      "service_checks": {
        "metadata_path": [
          "Missing data for required field."
        ]
      }
    }
  }
}
```

**Comment by nmanoogian** [comment]
I've been able to resolve most of the validation errors with trial-and-error but I'm a little stumped on the outstanding issue with logs:

```
Facet path `type` conflicts with other integrations: crest_data_systems_fortigate, jfrog_platform, pan.firewall, vault. Please fix this conflict or change this facet definition to:
groups:
  - Event
name: Type
path: type
source: log
```

File: doppler/datadog_checks/test/check.py
Comment (HadhemiDD) [inline doppler/datadog_checks/test/check.py:96]: The integration will always submit a critical service check, are you sure this is the desired behaviour or just a typo? 
Also, the namespace is test, this means all metrics / logs will be appended with the keyword test, the namespace should match the integration name "doppler"

File: doppler/tests/test_unit.py
Comment (HadhemiDD) [inline doppler/tests/test_unit.py:17]: What metrics are you asserting here? The integration do not submit any.

**Comment by HadhemiDD** [comment]
If the integration do not provide metrics or service checks, you can remove their section from the manifest.json file.
You also need to run this as the check is failing : `ddev validate ci --sync`

**Review by HadhemiDD — CHANGES_REQUESTED** [review]
Left some notes

File: doppler/assets/logs/doppler.yaml
Comment (thibaultkrebs) [inline doppler/assets/logs/doppler.yaml:46]: `preserveSource` should not be set to true, it will duplicate the property, we can simply keep the remapping one to false. Unless you have another good reason to do?
Same for usr.email.

File: doppler/assets/logs/doppler.yaml
Comment (thibaultkrebs) [inline doppler/assets/logs/doppler.yaml:15]: You need to remove the facetType here

**Comment by nmanoogian** [comment]
Thanks for the feedback, y'all! I've pushed up some fixes and re-requested your reviews.

File: doppler/README.md
Comment (emarsha94) [inline doppler/README.md:8]:
```suggestion
This integration sends Activity Logs from [Doppler][1] to Datadog, allowing you to monitor changes to your workplace.
```

Can you provide examples on changes?

Comment (nmanoogian) [reply doppler/README.md:8]: Absolutely 👍

File: doppler/assets/logs/doppler.yaml
Comment (thibaultkrebs) [inline doppler/assets/logs/doppler.yaml:38]: Actually I am not very clear why you need to have these 2 remappers? You are already submitting logs with these property set:
```
                    "usr": {
      				"name": "Nic Manoogian",
      				"email": "nic.manoogian@doppler.com"
      			},
      			"link": "https://dashboard.doppler.com/workplace/30bd9e7d881446ce1bf8/logs?id=wpmULcSfO9sNxKdVUeaO0HK2",
      			"type": "services.datadog.connect",
      			"title": "Activity Log: wpmULcSfO9sNxKdVUeaO0HK2",
      			"user": {
      				"name": "Nic Manoogian",
      				"profile_image_url": "https://www.gravatar.com/avatar/7f447a6cfe4f594f91c149b9c9ca872d?s=500&d=retro",
      				"email": "nic.manoogian@doppler.com",
      				"username": "nic"
      			},
  ```

Our system will already keep these values without the remapper. Is there some logs that will not include `usr` data?

Comment (nmanoogian) [reply doppler/assets/logs/doppler.yaml:38]: Hi @thibaultkrebs, that `usr` field is not include in the payload that we submit to Datadog, it's only present in the log because of the remapper

Comment (emarsha94) [reply doppler/assets/logs/doppler.yaml:38]: Hi @nmanoogian The sample should be a raw log before it is sent to Datadog. It sounds like you are saying that this sample is from Datadog after the logs went through the log pipeline. If that is the case, please update the sample to be a raw log sample before it is sent to Datadog.

Comment (emarsha94) [reply doppler/assets/logs/doppler.yaml:38]: @nmanoogian bumping the above - please update the sample. Once done, logs can review again.

**Review by emarsha94 — APPROVED** [review]

File: doppler/assets/oauth_clients.json
Comment (london-wharton) [inline doppler/assets/oauth_clients.json:11]:
```suggestion
    "id": "87fdb7f6-542c-11ef-abb5-da7ad0900002"
```
This is the correct uuid.

**Review by london-wharton — CHANGES_REQUESTED** [review]
Left a suggestion to update the Oauth client id. I believe this should fix the failing validations!

**Review by Kyle-Neale — APPROVED** [review]

File: doppler/assets/oauth_clients.json
Comment (nmanoogian) [reply doppler/assets/oauth_clients.json]: @london-wharton, @emarsha94: I'm having some trouble confirming that either of those UUIDs belongs to our production Datadog app. Can you confirm if it's associated with this app? https://app.datadoghq.com/apps/83469c40-caa8-11ee-8925-da7ad0900002/edit/oauth/8d0c113c-caa9-11ee-8b55-da7ad0900002

Comment (emarsha94) [reply doppler/assets/oauth_clients.json]: Correct - the published client id for that app is 87fdb7f6-542c-11ef-abb5-da7ad0900002

Comment (nmanoogian) [reply doppler/assets/oauth_clients.json]: Brilliant, thanks for the assist!

**Review by london-wharton — APPROVED** [review]
lgtm. Thanks for making that change!

File: doppler/manifest.json
Comment (emarsha94) [inline doppler/manifest.json:84]:
```suggestion
    "name": "Doppler",
```

File: doppler/assets/logs/doppler.yaml
Comment (thibaultkrebs) [inline doppler/assets/logs/doppler.yaml:36]: rechecking at the samples, I don't see any sample with this text property. Should all samples cover this property or could you provide at least one with this

Comment (nmanoogian) [reply doppler/assets/logs/doppler.yaml:36]: Great callout, I think I accidentally removed those when I reformatted the logs. I've added them back

**Comment by thibaultkrebs** [comment]
I have check in our test environment for logs files. You can find below 2 screenshots in our test env.

I will approve the PR as it is. But before merging, you can consider maybe one improvement:
You could create more remapper from user to usr properties, so it avoid having both `user` with 2 properties and `usr` with 2 properties. All can be remapped to `usr`.
image

Generated samples:
image
image

**Review by thibaultkrebs — APPROVED** [review]

**Comment by nmanoogian** [comment]
> I will approve the PR as it is. But before merging, you can consider maybe one improvement: You could create more remapper from user to usr properties, so it avoid having both `user` with 2 properties and `usr` with 2 properties. All can be remapped to `usr`

Good catch! I added remappers for those other two properties.

**Pull request merged**
